### PR TITLE
Fix ansible-test lint warnings in roles.

### PR DESCRIPTION
**Pull request by rjeffman**
This PR fixes several ansible-lint warnings on ansible-freeipa roles.

Please, see individual commits for details on the issues fixed.

### CODE REVIEW

File: .ansible-lint [hunk 1]
```diff
@@ -14,10 +14,11 @@ exclude_paths:
   - requirements-podman.yml
 
 kinds:
-  - playbook: '**/tests/**/test_*.yml'
-  - playbook: '**/playbooks/**/*.yml'
   - tasks: '**/tasks_*.yml'
   - tasks: '**/env_*.yml'
+  - tasks: '**/tests/azure/**/*.yml'
```
Comment (t-woerner) [inline .ansible-lint:19]: This line is good as long as ansible-lint does not enforce for example `ansible.builtin.script` use for the Azure templates.

File: roles/ipaserver/tasks/copy_external_cert.yml [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
-- name: Install - Copy "{{ item }}" "{{ inventory_hostname }}':/root/'{{ item | basename }}"
-  ansible.builtin.copy:
-    src: "{{ item }}"
-    dest: "/root/{{ item | basename }}"
-    mode: preserve
-    force: yes
-- name: Install - Extend ipaserver_external_cert_files with "/root/{{ item | basename }}"
-  ansible.builtin.set_fact:
-    ipaserver_external_cert_files: "{{ ipaserver_external_cert_files + [ '/root/' + (item | basename) ] }}"
+- name: Copy external certificate
```
Comment (t-woerner) [inline roles/ipaserver/tasks/copy_external_cert.yml:2]: This can be more than one.

File: roles/ipaserver/tasks/install.yml [hunk 3]
```diff
@@ -51,9 +52,7 @@
       >/dev/null
     when: ipaserver_firewalld_zone is defined
 
-  when: ipaserver_setup_firewalld | bool
-
-- name: Include tasks "{{ role_path }}/tasks/copy_external_cert.yml"
+- name: Copy external cert
```
Comment (t-woerner) [inline roles/ipaserver/tasks/install.yml:55]: This can be more than one.

File: roles/ipaserver/tasks/copy_external_cert.yml [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
+  vars:
+    __item_basename: "{{ item | basename }}"
+  block:
+  - name: Install - Initialize ipaserver_external_cert_files
+    ansible.builtin.set_fact:
+      ipaserver_external_cert_files: []
+    when: ipaserver_external_cert_files is undefined
+  - name: Install - Copy "{{ item ~ " " ~ inventory_hostname ~ ':/root/' ~ __item_basename }}"
```
Comment (t-woerner) [inline roles/ipaserver/tasks/copy_external_cert.yml:10]: When should `+` and when `~` be used?

File: roles/ipaserver/tasks/install.yml (hunk 3 above)
Comment (rjeffman) [reply roles/ipaserver/tasks/install.yml:55]: fixed.

File: roles/ipaserver/tasks/copy_external_cert.yml (hunk 4 above)
Comment (rjeffman) [reply roles/ipaserver/tasks/copy_external_cert.yml:10]: As discussed offline, change all string concatenation to use `+`.

File: .ansible-lint (hunk 1 above)
Comment (rjeffman) [reply .ansible-lint]: ok... this didn't work. I can only reproduce in Github workflow.

Added some more fine-grained control, but `tests/azure` is still being overriden, even if they are on the `exclude_paths`.

**Review by t-woerner — APPROVED** [review]
LGTM

**Pull request merged**